### PR TITLE
Fix dhcp infinite lease time

### PR DESCRIPTION
**Pull request by andrijapanicsb**
The previous setup of many hours would not work, due to some internal dnsmasq issues - lease was set correclty, but dnsmasq was setting the dhcp-renew-time (and rebind time) to less than 2 years from the date the lease was issued.

Using "infinite" as the value (instead of the number) works as expected - and (atm) the renew date is set to year 2088, etc..

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Screenshots (if appropriate):

Confirmation inside user VM of correct renew/rebind/expire time
image

Configuration inside VRs
image
Different one:
image

## How Has This Been Tested?

See screenshot above

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by andrijapanicsb** [comment]
> removing `lease = randint(870000, 870010)` is certainly OK. Not sure how `infinite` works. code lgtm

It works as one can imagine (I was intrusted by the dnsmasq ML guy to use that word and tested it works fine)

**Comment by andrijapanicsb** [comment]
@blueorangutan test

**Comment by svenvogel** [comment]
@andrijapanicsb LGTM and a small question. To ensure that the ip addresses are static in Cloudstack we use such a high lease time and they are only be changed from the operator. right?

**Comment by andrijapanicsb** [comment]
@svenvogel yes and not - I mean we can not change this (unless one edits the script/systemVM.iso) - here we are ensuring renew time is in far, far future ("infinite" as defined by the dnsmasq options) - thus effectively VM will never attempt to renew it's IP and thus will not be possible to fail to renew IP from the VR which has no knowledge of the issues lease (new VR that is created as part of the network restart with clean up - the /var/lib/misc/dnsmasq.leases is missing in this new VR - so VR will reject any DHCPRENEW request, and VM will eventually fail back to to  "initialisation" process with DHCPDISCOVER etc - but some OS (RHEL) will remove/delete their static routes when this happens...

**Comment by svenvogel** [comment]
> thus effectively VM will never attempt to renew it's IP and thus will not be possible to fail to renew IP from the VR which has no knowledge of the issued lease 

@andrijapanicsb Thanks for your explaination. 👍 I didn't have this sentence on my screen in particular.

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@weizhouapache @wido @GabrielBrascher @rhtyd @nvazquez and anybody else,...
I am going to merge this on no objections. It has quite an impact, so i'll leave it for a day for you all to have a chance to comment.

**Comment by andrijapanicsb** [comment]
(I've tested also on Windows 2019 that it gets the renewal time of "-1", as opposed to proper numbers - so will have to assume it's OK for this fellow as well)

(Also tested CentOS 8, more for fun than for anything else)

File: systemvm/debian/opt/cloud/bin/cs/CsDhcp.py [hunk 1]
```diff
@@ -174,18 +174,15 @@ def write_hosts(self):
...
         # so the infinite duration of lease does not cause any issues or garbage.
-        # There will be soon a PR which also regenerates the /var/lib/misc/dnsmasq.leases (active lease DB file)
-        # in the new VR (when restarting network with cleanup), which will help around RHEL edge cases (described above)
-        # for the VMs who are already running in productions systems with 30d lease time.
-        lease = randint(870000, 870010)
+        lease = 'infinite'
 
         if entry['default_entry']:
-            self.cloud.add("%s,%s,%s,%sh" % (entry['mac_address'],
+            self.cloud.add("%s,%s,%s,%s" % (entry['mac_address'],
```
Comment (DaanHoogland) [inline systemvm/debian/opt/cloud/bin/cs/CsDhcp.py]: pylint is not accepting this, @andrijapanicsb

File: systemvm/debian/opt/cloud/bin/cs/CsDhcp.py [hunk 2]
```diff
@@ -174,18 +174,15 @@ def write_hosts(self):
...
                                              entry['ipv4_address'],
                                              entry['host_name'],
                                              lease))
```
Comment (DaanHoogland) [inline systemvm/debian/opt/cloud/bin/cs/CsDhcp.py]: you need to unindent by one space as you made the base statement one char shorter as well. :P

**Comment by rhtyd** [comment]
I see @DaanHoogland has addressed his own review comment and travis is green. Tests LGTM.

**Pull request merged**